stdlib/public/core/AutoDiff.swift

```diff
@@ -163,10 +166,19 @@ public protocol Differentiable {
   /// All differentiable variables of this value.
   var allDifferentiableVariables: AllDifferentiableVariables { get set }
 
-  /// Moves `self` along the value space towards the given tangent vector. In
-  /// Riemannian geometry (mathematics), this represents an exponential map.
+  /// Moves `self` along the given direction. In Riemannian geometry,
+  /// this is equivalent to exponential map, which moves `self` on the
+  /// geodesic surface along the given tangent vector.
   mutating func move(along direction: TangentVector)
 
+  /// A tangent vector such that `move(along: zeroTangentVector)`
```

Comment:
The indentation here seems funny, how about:
```swift
  /// A tangent vector such that `move(along: zeroTangentVector)` will not
  /// modify `self`.
  ///
  /// - Note: `zeroTangentVector` can be `TangentVector.zero` in most cases,
  ///   but types whose tangent vectors depend on instance properties of
  ///   `self` need to provide a different implementation. For example, an
  ///   array’s zero tangent vector depends on the array’s `count`.
```

Reply:
Fixed.